jsonpickle/handlers.py

```diff
@@ -215,8 +215,24 @@ def flatten(self, obj, data):
 
     def restore(self, data):
         restore = self.context.restore
-        factory, args = [restore(i, reset=False) for i in data['__reduce__']]
-        return factory(*args)
+        res = [restore(i, reset=False) for i in data['__reduce__']]
+        factory = res.pop(0)
+        args = res.pop(0)
+        obj = factory(*args)
+
+        if len(res) >= 1:
```

Comment:
You might as well keep popping until res is empty, or use izip_longest, but fundamentally this is a good catch.